### PR TITLE
[CONTINT-3583] Start workloadmeta collectors in agent check commands

### PR DESCRIPTION
**Pull request by AliDatadog**
### What does this PR do?
This PR start workloadmeta collectors in agent check commands.

### Motivation
`agent check container` was broken.

### Additional Notes
N/A

### Describe how to test/QA your changes
Deploy the agent on kubernetes and run 
```
 k -n namespace exec pod -- agent check container -r --discovery-timeout 10 --discovery-min-instances 2
```
It should work

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/cli/subcommands/check/command.go [hunk 1]
```diff
@@ -141,9 +143,23 @@ func MakeCommand(globalParamsGetter func() GlobalParams) *cobra.Command {
 					LogParams:            logimpl.ForOneShot(globalParams.LoggerName, "off", true)}),
 				core.Bundle(),
 
+				// workloadmeta setup
+				collectors.GetCatalog(),
```
Comment (clamoriniere) [inline pkg/cli/subcommands/check/command.go:147]: dummy question: why this function call is not part of the fx.Provide(...) function?

File: pkg/cli/subcommands/check/command.go [hunk 2]
```diff
@@ -141,9 +143,23 @@ func MakeCommand(globalParamsGetter func() GlobalParams) *cobra.Command {
...
+				fx.Provide(func(config config.Component) workloadmeta.Params {
```
Comment (clamoriniere) [inline pkg/cli/subcommands/check/command.go:148]: I feel like we have this function logic now in different place, can/should we move it inside the workloadmeta package?

**Review by clamoriniere — COMMENTED** [review]
hi @AliDatadog,

this change should fix the issue that you describe in the PR, but could you explain more why the function was failing before so it will provide more context for other reviewers and also in the feature when we will be able to remove this "hack" 🙇

File: pkg/cli/subcommands/check/command.go (hunk 1 above)
Comment (zhuminyi) [reply pkg/cli/subcommands/check/command.go:147]: This function calls a collection of fx.provide and converts the collection of fx.options into a single option which is the same as single fx.provide returns.

**Comment by AliDatadog** [comment]
Hi @clamoriniere
The solution respects the current pattern of workloadmeta and its usage. The current desing decouples workloadmeta from its catalog of collectors and its init function.

- If those catalogs are not provided then the store will exist but no collector will be active. For the `agent check` command to work with the `container` check, we need them.
https://github.com/DataDog/datadog-agent/blob/f425dfa882dd9ca8533172c246ea047be1a40799/comp/core/workloadmeta/workloadmeta.go#L48-L58

- Moreover, its `initialization` function must be provided in `workloadmeta.Params` as it initializes the tagger.
https://github.com/DataDog/datadog-agent/blob/f425dfa882dd9ca8533172c246ea047be1a40799/cmd/agent/common/loader.go#L28-L49

I think that it will be modified as workloadmeta in a second iteration but I believe it's out of scope for this fix.

File: pkg/cli/subcommands/check/command.go (hunk 2 above)
Comment (AliDatadog) [reply pkg/cli/subcommands/check/command.go:148]: Indeed we do but I think that the current pattern is to define parameters before providing a module and there might be slight differences in the different parameters depending on the command. 

[Here](https://github.com/search?q=repo%3ADataDog%2Fdatadog-agent+workloadmeta.Params&type=code) are the different places were we define workloadmeta params.

Here is were we have exactly the same logic https://github.com/DataDog/datadog-agent/blob/f425dfa882dd9ca8533172c246ea047be1a40799/cmd/agent/subcommands/run/command.go#L308-L320 so we wouldn't benefit a lot from a factorization.
- It looks like the pattern is to define the parameters before the module

I chose to follow the current pattern but I am happy to receive any other proposal.

File: pkg/cli/subcommands/check/command.go [hunk 3]
```diff
@@ -141,9 +143,23 @@ func MakeCommand(globalParamsGetter func() GlobalParams) *cobra.Command {
...
+					var agentType workloadmeta.AgentType
+					if flavor.GetFlavor() == flavor.ClusterAgent {
```
Comment (zhuminyi) [inline pkg/cli/subcommands/check/command.go:150]: Where is flavor.setFlavor called?

Comment (zhuminyi) [reply pkg/cli/subcommands/check/command.go:150]: should be ok. it is default to node and will be reset in clusteragent.

File: pkg/cli/subcommands/check/command.go (hunk 2 above)
Comment (AliDatadog) [reply pkg/cli/subcommands/check/command.go:148]: I think there is a solution that could work by defining default params and setting the params as optional. Will take a deeper look

Comment (zhuminyi) [reply pkg/cli/subcommands/check/command.go:148]: Only two flavors are supported here: clusteragent and agent. Does cluster check use this function? If not, we can either move this piece of code to workloadmeta or simplify the condition check by removing cluster flavor.

Comment (AliDatadog) [reply pkg/cli/subcommands/check/command.go:148]: CLC runners have the flavor `NodeAgent` as well. I think we should indeed move it somewhere.

**Review by zhuminyi — APPROVED** [review]

**Review by hush-hush — APPROVED** [review]

**Pull request merged**